### PR TITLE
implement getFilteredWithExceeded method via cycles

### LINKED ISSUE #7: HW0. Implement UserMealsUtil.getFilteredWithExceeded method
Реализовать метод UserMealsUtil.getFilteredWithExceeded используя Java 8 Stream API.
Метод должен:
- возвращать только записи между startTime и endTime; 
- поле UserMealWithExceed.exceed должно показывать, превышает ли сумма калорий за весь день параметра метода caloriesPerDay.
Т.е UserMealWithExceed - это запись одной еды, но поле exceeded будет одинаково для всех записей за этот день.
- Оцените Time complexity вашего алгоритма, если он O(N*N)- попробуйте сделать O(N).

### PR DESCRIPTION
**Pull request by kortov**
closes #7

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,8 +34,16 @@
     </build>
 
     <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>RELEASE</version>
```
Comment (Shtramak) [inline pom.xml]: It's better to use concrete stable version of a library to not be confused in future

File: src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java [hunk 2]
```diff
@@ -17,4 +18,37 @@ public UserMealWithExceed(LocalDateTime dateTime, String description, int calori
...
+
+    public UserMealWithExceed(UserMeal userMeal, boolean exceed) {
+        this.dateTime = userMeal.getDateTime();
+        this.description = userMeal.getDescription();
+        this.calories = userMeal.getCalories();
+        this.exceed = exceed;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (Shtramak) [inline src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java]: UserMealWithExceed is an DTO object that is using just for transferring data from and to UserMeal. It shouldn't have any behaivior including equals & hashcode methods - https://en.wikipedia.org/wiki/Data_transfer_object

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 3]
```diff
@@ -3,29 +3,60 @@
...
-        return null;
+    public static List<UserMealWithExceed> getFilteredWithExceeded(List<UserMeal> mealList, LocalTime startTime, LocalTime endTime, int caloriesPerDay) {
+        List<UserMealWithExceed> resulList = new ArrayList<>();
+        Map<LocalDate, Integer> caloriesOnDate = new HashMap<>();
+
+        for (UserMeal userMeal : mealList) {
+            LocalDateTime localDateTime = userMeal.getDateTime();
+            LocalDate dateForMapKey = localDateTime.toLocalDate();
+            caloriesOnDate.merge(dateForMapKey, userMeal.getCalories(), Integer::sum);
+
```
Comment (Shtramak) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: redundant empty line

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 4]
```diff
@@ -3,29 +3,60 @@
...
+        }
+
+        if (caloriesOnDate.isEmpty()) {
+            return Collections.emptyList();
```
Comment (Shtramak) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: It's posible to provide one return for this method. This return is not a kind of "easy" situations (https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement)

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 5]
```diff
@@ -3,29 +3,60 @@
...
+        }
+
+        for (UserMeal userMeal : mealList) {
+            LocalDate mealDate = userMeal.getDateTime().toLocalDate();
+            if (caloriesOnDate.get(mealDate) == null && !isBetween(userMeal.getDateTime().toLocalTime(), startTime, endTime)) {
```
Comment (Shtramak) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: As I understood caloriesOnDate.merge() can't return null. In that case this if block is redundant https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#merge-K-V-java.util.function.BiFunction-

File: src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java (hunk 2 above)
Comment (kortov) [reply src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java]: Don't know. I don't sure about that point :) https://stackoverflow.com/questions/7252731/overriding-equals-method-in-dtos

Comment (Shtramak) [reply src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java]: hmm... Now I'm also not sure )

**Pull request merged**